spirekey/src/components/SendForm/SendForm.tsx

```diff
@@ -1,150 +1,323 @@
-    if (!isCAccount(data.sender)) return;
-    if (!process.env.NAMESPACE) return;
-    if (!device) return;
-
-    const caccount = decodeURIComponent(data.sender);
-    const command = await transfer({
-      sender: caccount,
-      receiver: data.receiver,
-      amount: data.amount,
-      gasPayer: data.gasPayer,
-      namespace: process.env.NAMESPACE,
-      networkId: network,
-      publicKey,
-    });
-
-    router.push(
-      `${process.env.WALLET_URL}/sign?transaction=${Buffer.from(
-        JSON.stringify(command),
-      ).toString('base64')}&returnUrl=${getReturnUrl('/submit')}`,
-    );
+    if (!account) throw new Error('No account connected');
+    if (!data.receiver) throw new Error('No receiver defined');
+    if (!data.networkId) throw new Error('No network selected');
+    try {
+      setIsLoading(true);
+      if (
+        !(await isCoinAccountExisting({
+          accountName: data.receiver,
+          networkId: data.networkId,
+          chainId: data.chainId as ChainId,
+        }))
+      ) {
+        setReceiverError('Account does not exist');
+        throw new Error('Account does not exist');
+      }
+      setReceiverError('');
+      const tx = getTransferTx({
+        amount: data.amount,
+        receiver: data.receiver,
+        account,
+        chainId: data.chainId as ChainId,
+      });
+      tx.setNetworkId(data.networkId);
+      const { transactions, isReady } = await sign(
+        [tx.createTransaction()],
+        [
+          {
+            accountName: account.accountName,
+            networkId: account.networkId,
+            chainIds: account.chainIds,
+            requestedFungibles: [
+              {
+                fungible: 'coin',
+                amount: data.amount + 0.1, // add 0.1 to account for gas fees
+              },
+            ],
+          },
+        ],
+      );
+      await isReady();
+      await Promise.all(
+        transactions.map(async (tx) => {
+          console.warn("DEBUGPRINT[5]: SendForm.tsx:170: tx=", tx)
+          const res = await l1Client.local(tx);
+          console.log('Preflight result', res);
+          const txDescriptor = await l1Client.submit(tx as ICommand);
+          const txRes = await l1Client.listen(txDescriptor);
+          setTx(txRes);
+        }),
+      );
+    } catch (e) {
+      if (e instanceof Error && e.message === 'Account does not exist') throw e;
+      console.warn('User canceled signin', e);
+    } finally {
+      setIsLoading(false);
+    }
   };
+  if (!account) return <div>Account not found</div>;
 
-  const amount = watch('amount');
-  const gasPayer = watch('gasPayer');
+  const amountProps = register('amount', {
+    valueAsNumber: true,
+    min: 0.000001,
+    max: parseFloat(account.balance),
+  });
 
-  if (!account)
+  if (tx)
     return (
-      <div>
-        Account not found <ButtonLink href="/">Go back home</ButtonLink>
-      </div>
+      <CardContainer hasPadding>
```

Comment:
I would suggest already using the updated card pattern components from @kadena/kode-ui and SpireKeyCardContentBlock (already has the icon preconfigured). I'm almost done converting the rest of them, there's just one styling issue with the title size on the following content blocks when they are stacked